lib/vm/src/instance.rs

```diff
@@ -109,6 +105,139 @@ pub(crate) struct Instance {
     vmctx: VMContext,
 }
 
+/// A collection of data about host envs.
+///
+/// Due to invariants in the `HostEnv` variant, this type should only
+/// be constructed by one of the given methods, not directly.
+#[derive(Debug)]
+pub enum ImportEnv {
```

Comment:
`VMImportFunctionEnv` might be a better name

Reply:
I left `VM` off of the type because of something Nick said a few days or weeks ago but what his mental model around `VM` types are: that they should be `repr(C)` and are directly interacted with by the VM.  I think there's a decent argument that this isn't a VM type and it doesn't even need to be defined in `wasmer_vm` actually, we could put it in `wasmer_types` because it doesn't contain anything VM-specific.